entoas/edges_test.go

```diff
@@ -0,0 +1,88 @@
+package entoas
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"entgo.io/contrib/entoas/serialization"
+	"entgo.io/ent/entc"
+	"entgo.io/ent/entc/gen"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWalk_CycleDepth(t *testing.T) {
```

Comment:
In general, don't test non-public APIs in unit tests.

Reply:
Rather use the public API to test the cases.

Unit tests serve as a stability guarantee and you don't want to do that for implementation details.

Reply:
I needed this function but the calling method was not done. I created this method TDD, this is why the test is there. Am I supposed to remove it, when done?